bibata-cursors: 2.0.3 -> 2.0.6; move to `pkgs/by-name` and fix `meta`

eclairevoyant opened the pull request:
## Description of changes

Move to `pkgs/by-name`, bump version, fix meta (remove `with lib;` as per #208242, and fix license)
Fixes #260804.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

kilianar commented:
Result of `nixpkgs-review pr 261532` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>bibata-cursors</li>
  </ul>
</details>

---
The relevant part of the log:
```
[Fail] Error occurred while processing build.toml:
Traceback (most recent call last):
  File "/nix/store/9g36qw8gwn171bmv0g6h7shqnz33n3ca-python3.10-clickgen-2.1.9/lib/python3.10/site-packages/clickgen/scripts/ctgen.py", line 156, in process
    cfg = parse_config_file(file, **kwargs)
  File "/nix/store/9g36qw8gwn171bmv0g6h7shqnz33n3ca-python3.10-clickgen-2.1.9/lib/python3.10/site-packages/clickgen/configparser.py", line 190, in parse_config>
    config = parse_toml_file(fp, **kwargs)
  File "/nix/store/9g36qw8gwn171bmv0g6h7shqnz33n3ca-python3.10-clickgen-2.1.9/lib/python3.10/site-packages/clickgen/configparser.py", line 148, in parse_toml_f>
    cursors = parse_cursors_section(d, config, **kwargs)
  File "/nix/store/9g36qw8gwn171bmv0g6h7shqnz33n3ca-python3.10-clickgen-2.1.9/lib/python3.10/site-packages/clickgen/configparser.py", line 113, in parse_cursor>
    x11_blob = open_blob(blobs, hotspot, x11_sizes, x11_delay)
  File "/nix/store/9g36qw8gwn171bmv0g6h7shqnz33n3ca-python3.10-clickgen-2.1.9/lib/python3.10/site-packages/clickgen/parser/__init__.py", line 22, in open_blob
    return parser(blob, hotspot, sizes, delay)  # type: ignore
  File "/nix/store/9g36qw8gwn171bmv0g6h7shqnz33n3ca-python3.10-clickgen-2.1.9/lib/python3.10/site-packages/clickgen/parser/png.py", line 83, in __init__
    super().__init__(blobs[0])
IndexError: list index out of range
@nix { "action": "setPhase", "phase": "installPhase" }
installing
cp: missing destination file operand after '/nix/store/5yzbv3m1x78sh6c3pr5ldapd3bxnbhmf-bibata-cursors-2.0.4/share/icons/'
```

eclairevoyant commented:
Meant to draft this, apologies

eclairevoyant committed:
bibata-cursors: move to `pkgs/by-name`

eclairevoyant committed:
bibata-cursors: 2.0.3 -> 2.0.6

updates build process based on some upstream breaking changes, and preserves pre/post-hook phases

eclairevoyant committed:
bibata-cursors: fix meta

quantenzitrone commented:
Result of `nixpkgs-review pr 261532` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bibata-cursors</li>
  </ul>
</details>
